### DISCUSSION TITLE
Is there a memory map yet?

### DISCUSSION THREAD

**Discussion by tkiller99999** [picocomputer/community#34] [Q&A]
I wanted to see if I can get the C64 roms to play nice and maybe run some C64 apps on this.  it would help a great deal to get a complete Memory map ( I see some info for various ram mapping)

**Comment by rumbledethumps** [comment]
It's all RAM except the last page $ff00 where the IO hardware is mapped.
https://github.com/picocomputer/rp6502-sdk/blob/main/rp6502.cfg

**Comment by tkiller99999** [comment]
So I can map rom code into the ram area and go from there. great..  well If
you know anyone who wants a Board (I got PCBWay to make them) I have 3 extra
ones.

 

I want to design a board that I can plug a cartridge (a c64 one) into and
read out and play the program from it.

 

 

 

 

 

Regards,

Ted Killmeyer

 

___________________________

Theodore Paul Killmeyer Jr.



(USS Nevada SSBN 733 Plank Owner)

Computer Engineer

Electrical Engineer

Senior Systems Architect

Skype:  Theodore.Killmeyer

Email:  ***@***.***> ***@***.***

Email:  ***@***.***> ***@***.***

Email:  ***@***.***> ***@***.***

Mobile: +1 (720) 937-0281

 

From: rumbledethumps ***@***.***> 
Sent: Saturday, August 19, 2023 2:58 PM
To: picocomputer/discussions ***@***.***>
Cc: tkiller99999 ***@***.***>; Author
***@***.***>
Subject: Re: [picocomputer/discussions] Is there a memory map yet?
(Discussion #34)

 

It's all RAM except the last page $ff00 where the IO hardware is mapped.
https://github.com/picocomputer/rp6502-sdk/blob/main/rp6502.cfg

-
Reply to this email directly, view it on GitHub
<https://github.com/orgs/picocomputer/discussions/34#discussioncomment-67705
89> , or unsubscribe
<https://github.com/notifications/unsubscribe-auth/ATCQLNVVK4326BQLTGANCOTXW
EZFJANCNFSM6AAAAAA3W2FK7A> .
You are receiving this because you authored the thread.
<https://github.com/notifications/beacon/ATCQLNT5Q3SGHXZIW4OF7D3XWEZFJA5CNFS
M6AAAAAA3W2FK7CWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3T
UL5UWJTQAM5HZ2.gif> Message ID:
***@***.***
***@***.***
.com> >

**Reply by brentward** [reply]
Since the full address space is mapped to RAM (excepting the last page 0xff00) you would have to change the address decoding logic in order to be able to address the ROM on a C64 cartridge. That would be doable but would probably require designing a new circuit board. 

Your bigger issue would be that the hardware is different. Most of the components aren't compatible. For example the RP6502 will not have a VIC II for video or a SID chip for sound. It uses the PIX VGA for video and the RIA for sound. Also none of the kernal calls will be available.

In order to design a new board that would allow you to just plug in a C64 rom and run it you'd need to mimic the memory map of a C64 and include all the hardware for a C64. It would essentially be building a new C64.

If you want to get a C64 program working, I think a more practical approach would be to port it and translate any video and audio access into reads and writes to the RIA and VGA and translate any calls to the C64 kernal into RIA api calls or otherwise handle them. Once you did that you could load the rom to ta USB drive and run it directly in ram without having to worry about redoing the address decoding.

**Comment by tkiller99999** [comment]
that sounds like a very goo idea....  I will work towards a programs that can scan through a c64 program and update the RIA and VGA into the code and if needed recompile theat...   or better yet just have it call a separate library for the calls
